strutil/shlex/shlex_test.go

```diff
@@ -157,3 +157,69 @@ func TestNastyReader(t *testing.T) {
 		t.Errorf("unexpected error")
 	}
 }
+
+func TestQuote(t *testing.T) {
+	safeUnquoted := "abcdefghijklmnopqrstuvwxyz" + "ABCDEFGHIJKLMNOPQRSTUVWXYZ" + "0123456789" + "@%_-+=:,./"
+	unicodeSample := "\xe9\xe0\xdf" // e + acute accent, a + grave, sharp s
+	unsafe := "\"`$\\!" + unicodeSample
+
+	type testInfo struct {
+		input    string
+		expected string
+	}
+
+	quoteTests := []testInfo{{
+		input:    ``,
+		expected: `''`,
+	}, {
+		input:    safeUnquoted,
+		expected: safeUnquoted,
+	}, {
+		input:    `foo bar xyz`,
+		expected: `'foo bar xyz'`,
+	}}
+
+	for _, u := range unsafe {
```

Comment:
Given we're doing table-driven tests anyway, I think it's clearer to specify a few cases with literals inline rather than having this loop which appends to `quoteTests` -- it's a bit hard to understand what's actually being tested as is.

Reply:
Thanks, I have updated the tests.